Modules/_ctypes/stgdict.c

```diff
@@ -367,40 +255,113 @@ PyCStructUnionType_update_stginfo(PyObject *type, PyObject *fields, int isStruct
     if (stginfo->flags & DICTFLAG_FINAL) {/* is final ? */
         PyErr_SetString(PyExc_AttributeError,
                         "_fields_ is final");
-        return -1;
+        goto error;
+    }
+
+    PyObject *layout_func = _PyImport_GetModuleAttrString("ctypes._layout",
+                                                          "get_layout");
+    if (!layout_func) {
+        goto error;
+    }
+    PyObject *kwnames = PyTuple_Pack(
+        2,
+        &_Py_ID(is_struct),
+        &_Py_ID(base));
+    if (!kwnames) {
+        goto error;
+    }
+    layout = PyObject_Vectorcall(
+        layout_func,
+        1 + (PyObject*[]){
+            NULL,
+            /* positional args */
+            type,
+            fields,
+            /* keyword args */
+            isStruct ? Py_True : Py_False,
+            baseinfo ? base : Py_None},
+        2 | PY_VECTORCALL_ARGUMENTS_OFFSET,
```

Comment:
I think it would be clearer with an intermediate variable (and we wouldn't need this cast).

Reply:
Hmm, I prefer this form mostly because it *doesn't* introduce a variable :)
(Technically, it's a compound literal, not a cast)

Reply:
Urgh, I forgot about those. Ok for keeping it.